crates/hyperdrive-math/src/lp.rs

```diff
@@ -600,6 +621,540 @@ impl State {
+        // we can't distribute excess idle since we ruled out the possibility
+        // of paying out the full maximum share reserves delta in step 3.
+        if share_proceeds >= max_share_reserves_delta {
+            return Err(eyre!(
+                "Share proceeds greater than or equal to the maximum share reserves delta."
+            ));
+        }
+
+        Ok((withdrawal_shares_redeemed, share_proceeds))
+    }
+
+    fn calculate_distribute_excess_idle_share_proceeds(
+        &self,
+        current_block_timestamp: U256,
+        active_lp_total_supply: FixedPoint, // just the number of lp tokens
+        withdrawal_shares_total_supply: FixedPoint,
+        max_share_reserves_delta: FixedPoint,
+        max_iterations: u64,
+    ) -> Result<FixedPoint> {
+        let net_curve_trade =
+            self.calculate_net_curve_trade_from_timestamp(current_block_timestamp)?;
+        // Calculate the LP total supply.
+        let lp_total_supply = active_lp_total_supply + withdrawal_shares_total_supply;
+        let starting_present_value = self.calculate_present_value(current_block_timestamp)?;
+
+        // NOTE: Round the initial guess down to avoid overshooting.
+        //
+        // We make an initial guess for Newton's method by assuming that the
+        // ratio of the share reserves delta to the withdrawal shares
+        // NOTE: Round the exponent up since this rounds the result up.
+        // outstanding is equal to the LP share price. In reality, the
+        // withdrawal pool should receive more than this, but it's a good
+        // starting point. The calculation is:
+        //
+        // x_0 = w * (PV(0) / l)
+        let mut share_proceeds =
+            withdrawal_shares_total_supply.mul_div_down(starting_present_value, lp_total_supply);
+
+        // If the pool is net neutral, the initial guess is equal to the final
+        // result.
+        if net_curve_trade == int256!(0) {
+            return Ok(share_proceeds);
+        }
+
+        // Proceed with Newton's method. The objective function, `F(x)`, is
+        // given by:
+        //
+        // F(x) = PV(x) * l - PV(0) * (l - w)
+        //
+        // Newton's method will terminate as soon as the current iteration is
+        // within the minimum tolerance or the maximum number of iterations has
+        // been reached.
+        let mut smallest_delta = I256::zero();
+        let mut closest_share_proceeds = fixed!(0);
+        let mut closest_present_value = fixed!(0);
+        for _ in 0..max(max_iterations, SHARE_PROCEEDS_MAX_ITERATIONS) {
+            // Clamp the share proceeds to the max share reserves delta since
+            // values above this threshold are always invalid.
+            share_proceeds = min(share_proceeds, max_share_reserves_delta);
+
+            // Simulate applying the share proceeds to the reserves.
+            //
+            // NOTE: If the updated reserves can't be calculated,  we can't
```

Comment:
extra space after comma

Reply:
also I'm not sure what this note is adding. Are you saying that newton's method cannot be performed if we can't calculate the updated reserves? Why would this happen?

Reply:
I see a similar note below on present value. I guess they're fine to leave but it seems unnecessary. Any time we use a `?` we're saying "if this can't be done, the loop breaks", no need to include a comment to that effect.

Reply:
yeah i agree, i'll remove it